mlflow/utils/rest_utils.py

```diff
@@ -186,6 +187,17 @@ def verify_rest_response(response, endpoint):
     return response
 
 
+def augmented_raise_for_status(response):
+    """Wrap the standard `requests.response.raise_for_status()` method and return reason"""
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if response.text:
+            raise HTTPError(f"{response.status_code} Error: {response.text}")
```

Comment:
```suggestion
            raise HTTPError(f"{e}. Response text: {e.response.text}")
```

Can we include the original error message?

```
# Now
requests.exceptions.HTTPError: 503 Error: Endpoint disabled due to the mlflow server running without `--serve-artifacts`. To enable artifacts server functionaltiy, run `mlflow server` with `--serve-artfiacts`
```

```
# Proposed
requests.exceptions.HTTPError: 503 Server Error: SERVICE UNAVAILABLE for url: http://localhost:5000/api/2.0/mlflow-artifacts/artifacts/experiments/0/e6fdde74db494268812a4286cff5100d/artifacts/a.txt. Response text: Endpoint disabled due to the mlflow server running without `--serve-artifacts`. To enable artifacts server functionaltiy, run `mlflow server` with `--serve-artfiacts`
```

Reply:
much better to have the server uri and the full statement in there. Updated!